ingestion-edge/ingestion_edge/publish.py

```diff
@@ -55,20 +56,22 @@ def _publish(topic: str, data: bytes, attrs: Dict[str, str]) -> str:
     worker.
     """
     try:
-        return client.publish(topic, data, **attrs).result()
+        return await async_wrap(client.publish(topic, data, **attrs))
     except PublishError:
         # Batch failed because pubsub permanently rejected at least one message
         # retry this message alone to determine if it was rejected
-        response = client.api.publish(
+        batch = client._batch_class(
+            autocommit=False,
+            client=client,
+            settings=client.batch_settings,
             topic=topic,
-            messages=[{'data': data, 'attributes': attrs}],
         )
-        if len(response.message_ids) != 1:
-            raise PublishError("message was not successfully published")
-        return response.message_ids[0]
+        future = batch.publish({'data': data, 'attributes': attrs})
+        batch.commit()  # spawns thread to commit batch
+        return await async_wrap(future)
 
 
-def flush(q: SQLiteAckQueue) -> Tuple[str, int]:
+async def flush(request: Request, q: SQLiteAckQueue) -> Tuple[str, int]:
```

Comment:
Is the return type annotation here still correct? I'd expect the new type to be a Response object, or maybe some sort of coroutine thing?

Reply:
ah, it is not correct. async functions return a coroutine when run, but I would expect `Tuple[str, int]` to be what you provide to `return`. The real problem is that this function now returns a `sanic.response.HTTPResponse` object.